### PR TITLE
Respect CFLAGS and LDFLAGS when compiling RefC

### PR DESCRIPTION
**Pull request by jeroendehaas**
This pull request allows users to pass additional flags to the C-compiler invoked by RefC. RefC now checks the environment for the presence of the (IDRIS2_)CFLAGS and (IDRIS2_)LDFLAGS variables, the longer versions taking priority. 

This is immediately useful to direct the C-compiler to the location of the gmp headers and libraries (which when using MacPorts reside in /opt/local/{include, lib}). Controlling this via the (IDRIS2_)CC variables will not work as compilation will fail due to the unused -L flag (which is not allowed under -Werror)

Version 0.5.1:

```
% IDRIS2_CG=refc idris2 -o whereisgmp gmp.idr
In file included from build/exec/whereisgmp.c:1:
In file included from /Users/jeroendehaas/.idris2/idris2-0.5.1/support/refc/runtime.h:3:
In file included from /Users/jeroendehaas/.idris2/idris2-0.5.1/support/refc/cBackend.h:8:
/Users/jeroendehaas/.idris2/idris2-0.5.1/support/refc/datatypes.h:8:10: fatal error: 'gmp.h' file not found
#include <gmp.h>
```

Trying via CC:

```
% IDRIS2_CG=refc CC="clang -L/opt/local/lib -I/opt/local/include" idris2 -o whereisgmp gmp.idr
clang: error: argument unused during compilation: '-L/opt/local/lib' [-Werror,-Wunused-command-line-argument]
```

After this PR:

```
% CFLAGS=-I/opt/local/include LDFLAGS=-L/opt/local/lib IDRIS2_CG=refc idris2 -o whereisgmp gmp.idr
```

### CODE REVIEW

**Comment by jeroendehaas** [comment]
Removed the trailing whitespace on around line 83. `lint/lint.py` now succeeds locally.

**Review by madman-bob — COMMENTED** [review]
This change could do with a test, to make sure that it doesn't get accidentally broken down the line.

File: src/Compiler/RefC/CC.idr [hunk 1]
```diff
@@ -12,13 +12,27 @@ import Libraries.Utils.Path
 
 %default total
 
+findEnv : String -> IO (Maybe String)
+findEnv varName =
+  do Nothing <- getEnv $ "IDRIS2" ++ varName
```
[the remark this answers is not in the document]
Comment (jeroendehaas) [reply src/Compiler/RefC/CC.idr]: Thanks for catching this. I just pushed a new version to address this. I am still looking at the structure of the test suite.

**Comment by jeroendehaas** [comment]
> This change could do with a test, to make sure that it doesn't get accidentally broken down the line.

Done. I've gone a different route and wrapped the C compiler with a shell script that logs if has been invoked before passing all its arguments to the actual C-compiler.

**Comment by benhormann** [comment]
OpenBSD doesn't have `env -u`, according to [env(1)](https://man.openbsd.org/env). I'd use `unset` instead (either before the command or in a subshell `(unset X; Y=y test-cmd)`).

**Comment by dunhamsteve** [comment]
On macos, I found the following will get things working:
```
export CPATH=/opt/homebrew/include:/opt/homebrew/lib
export IDRIS2_LIBS=/opt/homebrew/lib
```
The IDRIS2_LIBS value ends up in -L, and CPATH is used by clang for the header file search.

**Pull request closed without merging**